modules/apps/dispatch/dispatch-web/src/main/java/com/liferay/dispatch/web/internal/frontend/taglib/servlet/taglib/DispatchDetailsScreenNavigationCategory.java

```diff
@@ -14,76 +14,47 @@
 
 package com.liferay.dispatch.web.internal.frontend.taglib.servlet.taglib;
 
-import com.liferay.dispatch.constants.DispatchConstants;
-import com.liferay.dispatch.model.DispatchTrigger;
+import com.liferay.dispatch.web.internal.constants.DispatchScreenNavigationConstants;
 import com.liferay.frontend.taglib.servlet.taglib.ScreenNavigationCategory;
-import com.liferay.frontend.taglib.servlet.taglib.ScreenNavigationEntry;
-import com.liferay.frontend.taglib.servlet.taglib.util.JSPRenderer;
 import com.liferay.portal.kernel.language.Language;
 import com.liferay.portal.kernel.util.ResourceBundleUtil;
 
-import java.io.IOException;
-
 import java.util.Locale;
 import java.util.ResourceBundle;
 
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Reference;
 
 /**
  * @author Alessio Antonio Rendina
  */
 @Component(
-	property = {
-		"screen.navigation.category.order:Integer=10",
-		"screen.navigation.entry.order:Integer=10"
-	},
-	service = {ScreenNavigationCategory.class, ScreenNavigationEntry.class}
+	property = "screen.navigation.category.order:Integer=10",
+	service = ScreenNavigationCategory.class
 )
 public class DispatchDetailsScreenNavigationCategory
```

Comment:
@MrJohn1911 After splitting, the `Dispatch*ScreenNavigationCategory` classes share same logic of `getLabel(Locale)`, `getScreenNavigationKey()`, and the reference to `Language`. I see these classes are in 2 different modules. In your earlier changes, did you extract any base class to api module? If not, let's at least de-duplicate in dispatch-web.

Reply:
@MrJohn1911 Never mind for this.